src/pages/Script/index.tsx

```diff
@@ -0,0 +1,314 @@
+  const dispatch = useDispatch()
+  return (
+    <ScriptsTitleOverviewPanel>
+      <OverviewCard items={getScriptInfo(scriptInfo, dispatch)} hideShadow />
+    </ScriptsTitleOverviewPanel>
+  )
+}
+
+const LoadingState = ({ status, children }: { status: State.FetchStatus; children: ReactNode }) => {
+  switch (status) {
+    case 'Error':
+      return <Error />
+    case 'OK':
+      return <>{children}</>
+    case 'InProgress':
+    case 'None':
+    default:
+      return <Loading show />
+  }
+}
+
+export interface ScriptInfo {
+  scriptName: string
+  scriptType: string
+  typeId: number
+  codeHash: string
+  hashType: string
+  capacityOfDeployedCells: string
+  capacityOfReferringCells: string
+}
+
+export interface ScriptPageInfo<T> {
+  total: number
+  data: T[]
+  status: State.FetchStatus
+  page: number
+  size: number
+}
+
+const initScriptInfo: ScriptInfo = {
+  scriptName: '',
+  scriptType: '',
+  typeId: 0,
+  codeHash: '',
+  hashType: '',
+  capacityOfDeployedCells: '',
+  capacityOfReferringCells: '',
+}
+
+const initScriptPageInfo: ScriptPageInfo<any> = {
+  total: 0,
+  data: [],
+  status: 'None',
+  page: 1,
+  size: 10,
+}
+
+export const updatePageSection = (
+  fetchFunc: () => Promise<Response.Response<Response.Wrapper<any[]>> | null>,
+  setScriptInfo: Dispatch<SetStateAction<ScriptInfo>>,
+  setPagesInfo: Dispatch<SetStateAction<ScriptPageInfo<any>>>,
+  pagesInfo: ScriptPageInfo<any>,
+  pageDataName: 'ckbTransactions' | 'deployedCells' | 'referringCells',
+) => {
+  setPagesInfo({
+    ...pagesInfo,
+    status: 'InProgress',
+  })
+  fetchFunc()
+    .then(response => {
+      const data = response!.data as any
+      const meta = response!.meta as Response.Meta
+      setScriptInfo({
+        scriptName: data.scriptName,
+        scriptType: data.scriptType,
+        typeId: data.TypeId,
+        codeHash: data.codeHash,
+        hashType: data.hashType,
+        capacityOfDeployedCells: data.capacityOfDeployedCells,
+        capacityOfReferringCells: data.capacityOfReferringCells,
+      })
+      setPagesInfo({
+        ...pagesInfo,
+        data: data[pageDataName],
+        total: meta ? meta.total : 0,
+        status: 'OK',
+      })
+    })
+    .catch(error => {
+      const isEmpty = error && error.response && error.response.status === 404
+      setPagesInfo({
+        ...pagesInfo,
+        data: [],
+        total: 0,
+        status: isEmpty ? 'OK' : 'Error',
+      })
+    })
+}
+
+export const ScriptPage = () => {
+  const history = useHistory()
+  const { search, hash } = useLocation()
+  const [activeKey, setActiveKey] = useState<'transactions' | 'deployed_cells' | 'referring_cells'>('transactions')
+  const { codeHash } = useParams<{ codeHash: string }>()
+  const { hashType } = useParams<{ hashType: string }>()
+  const parsed = queryString.parse(search)
+  const [scriptInfo, setScriptInfo] = useState<ScriptInfo>(initScriptInfo)
```

Comment:
Why not use `useQuery`(https://react-query-v3.tanstack.com/reference/useQuery) to manage local states from HTTP requests, it supports cache and stale-while-revalidate

Reply:
> Why not use `useQuery`(https://react-query-v3.tanstack.com/reference/useQuery) to manage local states from HTTP requests, it supports cache and stale-while-revalidate

refactor to using react-query to request HTTP in each tab components